### PR TITLE
use fqdn for module

### PR DESCRIPTION
**Pull request by drewwells**
Used `gofmt` to rename `ssh3` to `github.com/francoismichel/ssh3`. The code hasn't been formatted so this resulted in a lot of whitespace changes.

### CODE REVIEW

**Comment by mpiraux** [comment]
Before I jump into the changes, it would be nice to find a way to enforce gofmt. Ideally it would be in a separated workflow. But also being able to run it before commit would be a good thing.

Maybe making a Makefile ? It could also setup a pre-commit script, but I'm not very knowledgeable on this part.
This could end up in a dedicated issue though.

**Review by mpiraux — COMMENTED** [review]
I eyeballed all the changes and they are fine

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -47,6 +47,10 @@ jobs:
         run: sudo ssh-keygen -b 4096 -t rsa -f /attacker_id_rsa -q -N ""
       - name: Install Go dependencies
         run: go get ./...
+      - name: Fmt code
+        run: |
+          go fmt ./...
+          git diff --exit-code
```
[the remark this answers is not in the document]
Comment (drewwells) [reply .github/workflows/build.yml:53]: Why isn't it running on pull requests? It should be required https://github.com/francoismichel/ssh3/blob/main/.github/workflows/build.yml#L2-L5

**Comment by drewwells** [comment]
> Before I jump into the changes, it would be nice to find a way to enforce gofmt. Ideally it would be in a separated workflow. But also being able to run it before commit would be a good thing.
> 
> Maybe making a Makefile ? It could also setup a pre-commit script, but I'm not very knowledgeable on this part. This could end up in a dedicated issue though.

I agree, the workflow should be turned into a makefile.

as far as enforcing `go fmt`, the code I add would do this. Most people configure their ide to use `go fmt`.

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (francoismichel) [reply .github/workflows/build.yml:53]: It is running on PRs indeed

File: cli/client/main.go [hunk 2]
```diff
@@ -648,7 +642,7 @@ func mainWithStatusCode() int {
 	if err != nil {
 		log.Fatal().Msgf("%s", err)
 	}
-	req.Proto = "ssh3"
+	req.Proto = "github.com/francoismichel/ssh3"
```
Comment (francoismichel) [inline cli/client/main.go:645]: Not sure about that one, though, we're setting the `:protocol` HTTP/3 pseudo-header here like it is done fir [webtransport](https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3-08#name-creating-a-new-session), or [masque](https://datatracker.ietf.org/doc/html/draft-ietf-masque-connect-udp-15#name-http-2-and-http-3-requests). So I would let that set to `ssh3` for now.

File: linux_server/authorized_identities.go [hunk 3]
```diff
@@ -53,7 +54,7 @@ func (i *PubKeyIdentity) Verify(genericCandidate interface{}, base64Conversation
 			return nil, fmt.Errorf("unsupported signature algorithm '%s' for %T", unvalidatedToken.Method.Alg(), i)
 		},
 			jwt.WithIssuer(i.username),
-			jwt.WithSubject("ssh3"),
+			jwt.WithSubject("github.com/francoismichel/ssh3"),
```
Comment (francoismichel) [inline linux_server/authorized_identities.go:57]: same as my above comment, if another implem appears, we would not we interoperable, so I would keep `ssh3` for now

File: server.go [hunk 4]
```diff
@@ -130,7 +130,7 @@ func (s *Server) GetHTTPHandlerFunc(ctx context.Context) AuthenticatedHandlerFun
 
 	return func(authenticatedUsername string, newConv *Conversation, w http.ResponseWriter, r *http.Request) {
 		log.Info().Msgf("got request: method: %s, URL: %s", r.Method, r.URL.String())
-		if r.Method == http.MethodConnect && r.Proto == "ssh3" {
+		if r.Method == http.MethodConnect && r.Proto == "github.com/francoismichel/ssh3" {
```
Comment (francoismichel) [inline server.go:133]: Same here, I would keep `ssh3` to keep interoperability

**Review by francoismichel — CHANGES_REQUESTED** [review]
Thank for the formatting, I wish I did that before releasing the implem publicly.

A few important requested changes needed to keep interoperability though. I can commit it myself if you want.

File: client.go [hunk 5]
```diff
@@ -299,7 +298,7 @@ func buildJWTBearerToken(signingMethod jwt.SigningMethod, key interface{}, usern
 		"iss":       username,
 		"iat":       jwt.NewNumericDate(time.Now()),
 		"exp":       jwt.NewNumericDate(time.Now().Add(10 * time.Second)),
-		"sub":       "ssh3",
+		"sub":       "github.com/francoismichel/ssh3",
```
Comment (drewwells) [inline client.go:301]:
```suggestion
		"sub":       "ssh3",
```

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (drewwells) [reply .github/workflows/build.yml:53]: hmm, I still don't see PR tests running on this PR. Are they enabled in the settings?
https://github.com/francoismichel/ssh3/settings/actions

[the remark this answers is not in the document]
Comment (francoismichel) [reply .github/workflows/build.yml:53]: It runs, but requires approval to run tests for PRs from first-time contributors.
They just ran on this PR a few seconds ago:

image

**Review by francoismichel — APPROVED** [review]
Reviewed every change, looks good to me. Thanks for the reformat !

**Pull request merged**